includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -394,6 +394,28 @@ private function adjust_and_replace_nodes_in_array_map( $node_lists ) {
 		}
 	}
 
+	/**
+	 * Add 'srcset' attribute to image node if lightbox is enabled for that.
+	 *
+	 * @param Element $node Image node.
+	 *
+	 * @return void
+	 */
+	private function set_srcset_for_lightbox_images( Element $node ) {
+
+		if (
+			! $node->hasAttribute( Attribute::SRCSET )
+			&&
+			$node->hasAttribute( 'data-amp-lightbox' )
+			&&
```

Comment:
This should be OR condition? 
If the user doesn't opt-in for Lightbox but opt-in for carousel it should also generate srcset otherwise it will show distorted images in carousel

Reply:
https://user-images.githubusercontent.com/1132541/163835692-f2cb8198-457e-4812-a18c-89dfc937fd99.mov

Reply:
Here we are setting `srcset` because we need to show full size image in lightbox.